core-api/src/main/java/com/optimizely/ab/config/Experiment.java

```diff
@@ -173,6 +173,101 @@ public boolean isLaunched() {
+        if (audiencesMap == null ) return "\"" + audienceId + "\"";
+        String audienceName = "\"" + audiencesMap.get(audienceId) + "\"";
+        return audiencesMap.get(audienceId) != null ? audienceName : "\"" + audienceId + "\"";
+    }
+
+    private String getOperandOrAudienceId(Condition condition, Map<String, String> audiencesMap){
+        String operand = "";
+        if (condition != null) {
+            if (condition instanceof AndCondition) {
+                operand = "AND";
+            } else if (condition instanceof NotCondition) {
+                operand = "NOT";
+            } else if (condition instanceof OrCondition) {
+                operand = "OR";
+            } else if (condition instanceof AudienceIdCondition) {
+                String audienceName = this.getNameFromAudienceId(((AudienceIdCondition<?>) condition).getAudienceId(),
+                    audiencesMap);
+                operand = audienceName;
+            }
+        }
+        return operand;
+    }
+
+    public String serialize(Condition condition, Map<String, String> audiencesMap) {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("");
+        List<Condition> conditions;
+
+        String operand = this.getOperandOrAudienceId(condition, audiencesMap);
+        switch (operand){
+            case ("AND"):
+                conditions = ((AndCondition<?>) condition).getConditions();
+                stringBuilder.append(this.getNameOrNextCondition(operand, conditions, audiencesMap));
+                break;
+            case ("OR"):
+                conditions = ((OrCondition<?>) condition).getConditions();
+                stringBuilder.append(this.getNameOrNextCondition(operand, conditions, audiencesMap));
+                break;
+            case ("NOT"):
+                stringBuilder.append(operand + " ");
+                Condition notCondition = ((NotCondition<?>) condition).getCondition();
+                if (notCondition instanceof AudienceIdCondition) {
+                    stringBuilder.append(serialize(notCondition, audiencesMap));
+                } else {
+                    stringBuilder.append("(" + serialize(notCondition, audiencesMap) + ")");
+                }
+                break;
+            default:
+                stringBuilder.append(operand);
+                break;
+        }
+
+        return stringBuilder.toString();
+    }
+
+    public String getNameOrNextCondition(String operand, List<Condition> conditions, Map<String, String> audiencesMap) {
+        StringBuilder stringBuilder = new StringBuilder();
+        int ctr = 0;
+        if (conditions.isEmpty()) {
+            return "";
+        } else if (conditions.size() == 1) {
+            return serialize(conditions.get(0), audiencesMap);
+        } else {
+            for (Condition con : conditions) {
+                ctr++;
```

Comment:
ctr ?

Reply:
ctr is short for counter, will rename to something more meaningful